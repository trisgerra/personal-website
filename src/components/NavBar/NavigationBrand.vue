<template>
  <div class="navbar-brand">
    <router-link class="menu-item" to="/">
      <div class="navbar-item navbar-logo">
        <img src="../../assets/logo.png" alt="ADG Official" height="38" />
      </div>
    </router-link>
    <a
      role="button"
      class="navbar-burger"
      aria-label="menu"
      aria-expanded="false"
      @click="this.switchActive">
      <span aria-hidden="true"></span>
      <span aria-hidden="true"></span>
      <span aria-hidden="true"></span>
    </a>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { IRoute, routes } from '../../router';

@Component
export default class NavigationBrand extends Vue {
  @Prop() switchActive!: Function;
}
</script>

<style scoped lang="scss">

.navbar-logo {
  height: 100%;
  background-color: #151616;
  border-left: 3px solid #151616;
  &:hover {
    background-color: #151616;
    border-left: 3px solid #ff0000;
  }
}
</style>
