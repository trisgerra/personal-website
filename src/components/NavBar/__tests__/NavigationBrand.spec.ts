import { Wrapper } from '@vue/test-utils';
import NavigationBrand from '../NavigationBrand.vue';
import createWrapper from '@/utils/CreateWrapper';

describe('NavigationBrand', () => {
  it('should render as expected', () => {
    const wrapper: Wrapper<NavigationBrand> = createWrapper(NavigationBrand);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('contains logo', () => {
    const wrapper: Wrapper<NavigationBrand> = createWrapper(NavigationBrand);
    expect(wrapper.findAll('.navbar-logo').length).toBe(1);
  });

  it('contains burger menu for smaller device', () => {
    const wrapper: Wrapper<NavigationBrand> = createWrapper(NavigationBrand);
    expect(wrapper.findAll('.navbar-burger').length).toBe(1);
  });

  it('click on burger menu should trigger switchActive function', () => {
    const switchActiveSpy = jest.fn();
    const wrapper: Wrapper<NavigationBrand> = createWrapper(NavigationBrand, {
      propsData: {
        switchActive: switchActiveSpy,
      },
    });
    wrapper.find('.navbar-burger').trigger('click');
    expect(switchActiveSpy).toHaveBeenCalledTimes(1);
  });
});
