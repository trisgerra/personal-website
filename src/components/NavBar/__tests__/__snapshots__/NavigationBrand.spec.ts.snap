// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBrand should render as expected 1`] = `
<div
  class="navbar-brand"
>
  <a
    class="menu-item router-link-exact-active router-link-active"
    href="#/"
  >
    <div
      class="navbar-item navbar-logo"
    >
      <img
        alt="ADG Official"
        height="38"
        src="../../assets/logo.png"
      />
    </div>
  </a>
   
  <a
    aria-expanded="false"
    aria-label="menu"
    class="navbar-burger"
    role="button"
  >
    <span
      aria-hidden="true"
    />
     
    <span
      aria-hidden="true"
    />
     
    <span
      aria-hidden="true"
    />
  </a>
</div>
`;
